advanced/Scripts/list.sh

```diff
@@ -88,175 +88,144 @@ HandleOther() {
-            # Remove it from the other one
-            echo -e "  ${INFO} Removing $domain from regex list..."
-            local lineNumber
-            lineNumber=$(grep -Fnx "$domain" "${list}" | cut -f1 -d:)
-            sed -i "${lineNumber}d" "${list}"
-            reload=true
-        else
-            if [[ "${verbose}" == true ]]; then
-                echo -e "  ${INFO} ${domain} does not exist in regex list, no need to remove!"
-            fi
+        reload=true
+        # Remove it from the current list
+        local timestamp
+        timestamp="$(date --utc +'%s')"
+        sqlite3 "${gravityDBfile}" "DELETE FROM ${list} WHERE ${sqlitekey} = \"${domain}\";"
+    else
+        if [[ "${verbose}" == true ]]; then
+            echo -e "  ${INFO} ${1} does not exist in ${list}, no need to remove!"
         fi
     fi
 }
 
-# Update Gravity
-Reload() {
-    echo ""
-    pihole -g --skip-download "${type:-}"
-}
-
 Displaylist() {
-    if [[ -f ${listMain} ]]; then
-        if [[ "${listMain}" == "${whitelist}" ]]; then
-            string="gravity resistant domains"
-        else
-            string="domains caught in the sinkhole"
-        fi
-        verbose=false
-        echo -e "Displaying $string:\n"
+    local list listname count num_pipes domain enabled status
+
+    if [[ "${listType}" == "regex" ]]; then
+        listname="regex filters list"
+    else
+        # Whitelist / Blacklist
+        listname="${listType}"
+    fi
+    data="$(sqlite3 "${gravityDBfile}" "SELECT * FROM ${listType};" 2> /dev/null)"
+
+    if [[ -z $data ]]; then
+        echo -e "Not showing empty ${listname}"
+    else
+        echo -e "Displaying ${listname}:"
         count=1
-        while IFS= read -r RD || [ -n "${RD}" ]; do
-            echo "  ${count}: ${RD}"
+        while IFS= read -r line
+        do
+            # Count number of pipes seen in this line
+            # This is necessary because we can only detect the pipe separating the fields
+            # from the end backwards as the domain (which is the first field) may contain
+            # pipe symbols as they are perfectly valid regex filter control characters
+            num_pipes="$(grep -c "^" <<< "$(grep -o "|" <<< "${line}")")"
+
+            # Extract domain and enabled status based on the obtained number of pipe characters
+            domain="$(cut -d'|' -f"-$((num_pipes-2))" <<< "${line}")"
+            enabled="$(cut -d'|' -f"$((num_pipes-1))" <<< "${line}")"
+
+            # Translate boolean status into human readable string
+            if [[ "${enabled}" -eq 1 ]]; then
+                status="enabled"
+            else
+                status="disabled"
+            fi
+
+            echo "  ${count}: ${domain} (${status})"
             count=$((count+1))
-        done < "${listMain}"
-    else
-        echo -e "  ${COL_LIGHT_RED}${listMain} does not exist!${COL_NC}"
+        done <<< "${data}"
+        exit 0;
     fi
-    exit 0;
```

Comment:
This exit seems to have been fine where it was, is there a reason for moving it?